### PR TITLE
feat: added layer controller base logic + fixed error handler

### PR DESCRIPTION
**Pull request by CL-SHLOMIKONCHA**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                  |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✖                                                                        |
| Chore            | ✔                                                                       |

Related issues: #XXX , #XXX ...
Closes #XXX ...

Further  information:

### CODE REVIEW

File: src/resourceName/models/layersManager.ts [hunk 1]
```diff
@@ -1,17 +1,50 @@
...
+    const newLayer: MapProxyLayer = {
+      name: layerRequest.name,
+      title: layerRequest.description,
+      sources: [layerRequest.name],
+    };
+    jsonDocument.caches[layerRequest.name] = newCache;
+    jsonDocument.layers.push(newLayer);
+
+    const yamlContent = convertJsonToYaml(jsonDocument);
+    replaceYamlFileContent(yamlContent);
```
Comment (asafMasa) [inline src/resourceName/models/layersManager.ts:48]: add log on completion

File: src/resourceName/models/layersManager.ts [hunk 2]
```diff
@@ -1,17 +1,50 @@
...
+
   public getLayer(): LayerPostRequest {
     this.logger.log('info', 'get layer request');
     return mockLayer;
   }
-  public addLayer(newLayer: LayerPostRequest): LayerPostRequest {
-    this.logger.log('info', `add layer request: ${newLayer.name}`);
-    return mockLayer;
+
+  public addLayer(layerRequest: LayerPostRequest): void {
```
Comment (asafMasa) [inline src/resourceName/models/layersManager.ts:20]: Add Tests: take reference from here:
https://github.com/MapColonies/discrete-overseer-creator/blob/master/tests/unit/layers/models/progressModel.spec.ts#L42

**Review by asafMasa — CHANGES_REQUESTED** [review]
🚀 Good Job

File: src/common/interfaces.ts [hunk 3]
```diff
@@ -16,8 +18,43 @@ export interface OpenApiConfig {
...
+  cache: {
+    grids: string[];
+    request_format: string;
+    upscale_tiles: number;
+    type: string;
+    directory_layout: string;
+  };
+}
+
+export interface MapProxyJsonDocument {
```
[the remark this answers is not in the document]
Comment (CL-SHLOMIKONCHA) [reply src/common/interfaces.ts]: fixed on 2b40b93

File: src/resourceName/models/layersManager.ts (hunk 1 above)
Comment (CL-SHLOMIKONCHA) [reply src/resourceName/models/layersManager.ts]: fixed on 19ee851

File: src/resourceName/models/layersManager.ts (hunk 2 above)
Comment (CL-SHLOMIKONCHA) [reply src/resourceName/models/layersManager.ts]: fixed on c72ad19

**Review by asafMasa — APPROVED** [review]

**Pull request merged**